### PR TITLE
unix: allow uv_getnameinfo to be cancelled

### PR DESCRIPTION
**Pull request by saghul**
Looks like this one slipped on the initial implementation.

/cc @indutny

### CODE REVIEW

**Comment by zerhacken** [comment]
@saghul oups, nice spotted.

File: test/test-threadpool-cancel.c [hunk 1]
```diff
@@ -202,6 +210,44 @@ TEST_IMPL(threadpool_cancel_getaddrinfo) {
...
+  loop = uv_default_loop();
+  saturate_threadpool();
+
+  r = uv_getnameinfo(loop, reqs + 0, getnameinfo_cb, (const struct sockaddr*)&addr4, 0);
+  ASSERT(r == 0);
+
+  r = uv_getnameinfo(loop, reqs + 1, getnameinfo_cb, (const struct sockaddr*)&addr4, 0);
+  ASSERT(r == 0);
+
+  r = uv_getnameinfo(loop, reqs + 2, getnameinfo_cb, (const struct sockaddr*)&addr4, 0);
```
Comment (indutny) [inline test/test-threadpool-cancel.c:233]: `ASSERT(0 == ...)`?

**Comment by indutny** [comment]
LGTM

File: test/test-threadpool-cancel.c (hunk 1 above)
Comment (saghul) [inline test/test-threadpool-cancel.c:233]: Makes the line too long. Also, this test is copied verbatim from the getaddrinfo one. The new ones I do write them like that :-)

**Comment by saghul** [comment]
Thanks, will merge!

**Comment by saghul** [comment]
Landed!

**Pull request merged**